mosquitto/plan.sh

```diff
@@ -0,0 +1,31 @@
+pkg_name=mosquitto
+pkg_origin=core
+pkg_version="1.4.14"
+pkg_upstream_url="https://mosquitto.org"
+pkg_description="An Open Source MQTT v3.1/v3.1.1 Broker"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=('EPL-1.0' 'Eclipse Distribution License - v 1.0')
+pkg_source="http://mosquitto.org/files/source/mosquitto-${pkg_version}.tar.gz"
+pkg_shasum="156b1fa731d12baad4b8b22f7b6a8af50ba881fc711b81e9919ec103cf2942d1"
+pkg_deps=(core/glibc core/gcc-libs core/openssl core/c-ares core/util-linux)
```

Comment:
I was able to build without `core/gcc-libs`. Not sure if that is relevant though.

Reply:
I'll try and see if the `libmosquittopp.so` is linked properly without `core/gcc-libs`

Reply:
This breaks linking even if the build passes:

```
[4][default:/src:0]# ldd /hab/pkgs/rsertelon/mosquitto/1.4.14/20171112104129/lib/libmosquittopp.so
	linux-vdso.so.1 (0x00007ffc40b35000)
	libmosquitto.so.1 => /hab/pkgs/rsertelon/mosquitto/1.4.14/20171112104129/lib/libmosquitto.so.1 (0x00007f9e4fcf6000)
	libstdc++.so.6 => not found
	libm.so.6 => /hab/pkgs/core/glibc/2.22/20170513201042/lib/libm.so.6 (0x00007f9e4f9f8000)
	libgcc_s.so.1 => not found
	libc.so.6 => /hab/pkgs/core/glibc/2.22/20170513201042/lib/libc.so.6 (0x00007f9e4f654000)
	librt.so.1 => /hab/pkgs/core/glibc/2.22/20170513201042/lib/librt.so.1 (0x00007f9e4f44c000)
	libssl.so.1.0.0 => /hab/pkgs/core/openssl/1.0.2l/20171101045054/lib/libssl.so.1.0.0 (0x00007f9e4f1da000)
	libcrypto.so.1.0.0 => /hab/pkgs/core/openssl/1.0.2l/20171101045054/lib/libcrypto.so.1.0.0 (0x00007f9e4ed87000)
	libpthread.so.0 => /hab/pkgs/core/glibc/2.22/20170513201042/lib/libpthread.so.0 (0x00007f9e4eb6a000)
	libcares.so.2 => /hab/pkgs/core/c-ares/1.13.0/20171107214749/lib/libcares.so.2 (0x00007f9e4e959000)
	/hab/pkgs/core/glibc/2.22/20170513201042/lib64/ld-linux-x86-64.so.2 (0x00007f9e5010d000)
	libdl.so.2 => /hab/pkgs/core/glibc/2.22/20170513201042/lib/libdl.so.2 (0x00007f9e4e755000)
	libz.so.1 => /hab/pkgs/core/zlib/1.2.8/20170513201911/lib/libz.so.1 (0x00007f9e4e53a000)

```

Reply:
Ah, yeah, I'm very new to Habitat in general. My plan doesn't even set the `pkg_lib_dirs` and I only set `(bin)` for `pkg_bin_dirs`. No idea how it's able to run without it's libraries.

```pkg_name=mosquitto
pkg_origin=jerry
pkg_version="1.4.14"
pkg_maintainer="Jerry Aldrich <jerryaldrichiii@gmail.com>"
pkg_license=('Apache-2.0')
pkg_filename="${pkg_name}-${pkg_version}.tar.gz"
pkg_source="http://mosquitto.org/files/source/$pkg_filename"
pkg_shasum="156b1fa731d12baad4b8b22f7b6a8af50ba881fc711b81e9919ec103cf2942d1"
pkg_build_deps=(core/make core/gcc)
pkg_deps=(core/glibc core/openssl core/c-ares core/util-linux)
pkg_bin_dirs=(bin)
pkg_svc_run="mosquitto -c $pkg_svc_config_path/mosquitto.conf"
pkg_exports=(
  [port]=cfg.port
)

do_build() {
  make
}

do_install() {
  source_dir=$HAB_CACHE_SRC_PATH/${pkg_dirname}
  
  cp $source_dir/src/mosquitto ${pkg_prefix}/bin/
  cp $source_dir/src/mosquitto_passwd ${pkg_prefix}/bin/
}
```

Reply:
Actually, the `mosquitto` binary works just fine without the libs. The dependency is required for those who might package a C++ program that makes use of `libmosquittopp.so` ;) (and maybe for `mosquitto_sub` and `mosquitto_pub`)

Reply:
Awesome, that makes sense! Thanks!